include/Basic/SourceManager.hpp

```diff
@@ -166,6 +166,21 @@ class SourceManager {
     unsigned getSpellingColumnNumber(SourceLocation loc) const;
     unsigned getSpellingLineNumber(SourceLocation loc) const;
 
+    /// @brief Get the content of the buffer for a specific location
+    /// @param loc The source location
+    /// @return The content of the buffer as a StringRef
+    llvm::StringRef getBufferData(SourceLocation loc) const;
+
+    /// @brief Get the offset of the start of the line containing the location
+    /// @param loc The source location
+    /// @return The offset of the start of the line
+    size_t getLineStart(SourceLocation loc) const;
```

Comment:
If the return value is supposed to be a source location, the type should be SourceLocation